src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -749,4 +776,38 @@ public override IReadOnlyList<Type> GetParameterTypes(string methodName)
 
         return null;
     }
+
+    private static Activity? CreateActivity(IServiceProvider serviceProvider, string methodName)
+    {
+        if (serviceProvider.GetService<ActivitySource>() is ActivitySource activitySource
+            && activitySource.HasListeners())
+        {
+            var requestContext = Activity.Current?.Context;
+            // Get off the parent span.
+            // This is likely the Http Request span and we want Hub method invocations to not be collected under a long running span.
+            Activity.Current = null;
+
+            var activity = activitySource.CreateActivity($"{typeof(THub).Name}/{methodName}", ActivityKind.Server, parentId: null,
+                links: requestContext.HasValue ? [new ActivityLink(requestContext.Value)] : null);
+
+            if (activity is not null)
+            {
+                // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#server-attributes
+                activity.AddTag("rpc.method", methodName);
+                activity.AddTag("rpc.system", "signalr");
+                activity.AddTag("rpc.service", typeof(THub).Name);
```

Comment:
Can a hub name be changed with an attribute or some other API? This value should take it into account. From spec notes:

> This is the logical name of the service from the RPC interface perspective, which can be different from the name of any implementing class.

Also, I think this value should probably include the namespace. Does SignalR qualify the hub name with a namespace? e.g. can you have two Product hubs in different namespaces?